docs/native-code.md

```diff
@@ -0,0 +1,122 @@
+
+### Isolation
+The app runs inside of an app container - a type of sandbox. This provides apps with a secure way to install, access system resources like the filesystem, and lets the system manage their lifetime (e.g. suspending the app when it isn't on the foreground). This means that by default an RNW app cannot access arbitrary filesystem locations, start arbitrary processes, etc. UWP apps that need to access these kinds of capabilities may be able to do so via [App capability declarations](https://docs.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations).
+
+### Packaging
+
+They are signed and packaged. [Packaging](https://docs.microsoft.com/en-us/windows/uwp/packaging/) is a mechanism through which an app and its dependencies acquire an identity, which is used to determine whether API calls that require system capabilities (e.g. filesystem access) should succeed or not.
+
+### Distribution
+They can be distributed, installed and updated in the following ways:
+  - via [the Microsoft Store](https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/desktop-to-uwp-distribute).
+  - via [your private Store](https://docs.microsoft.com/en-us/microsoft-store/distribute-apps-to-your-employees-microsoft-store-for-business) if you are a business or educational organization. See also [Distribute LOB apps to enterprises](https://docs.microsoft.com/en-us/windows/uwp/publish/distribute-lob-apps-to-enterprises).
+  - using [App Installer](https://docs.microsoft.com/en-us/windows/msix/app-installer/installing-windows10-apps-web).
+  - It's worth noting that you cannot just "_copy an exe_" as the app package contains more than just the main executable, including an app manifest, assets, dependent framework libraries, etc.
+- UWP apps submitted to the store must pass [Windows App Certification Kit](https://docs.microsoft.com/en-us/windows/uwp/debug-test-perf/windows-app-certification-kit) (WACK) validation.
+- UWP apps written in C# or other managed languages submitted to the store must be built using the [.NET Native toolchain](https://docs.microsoft.com/en-us/dotnet/framework/net-native/). This is the default when building C# apps in Release mode, **but not in Debug**, so apps built in Debug will be rejected by the Store.
+
+
+### Use of non-WinRT libraries
+Any libraries you use should be built as WinRT components. In other words, you cannot easily link libraries built for win32 desktop apps without additional work.
+  - C++/CX is a dialect of C++ that allows writing UWP apps, however this is **not supported** for writing a RNW app. The article [How to use existing C++ code in a Universal Windows Platform app](https://docs.microsoft.com/en-us/cpp/porting/how-to-use-existing-cpp-code-in-a-universal-windows-platform-app) talks about how to consume non-WinRT libraries in a WinRT context using C++/CX, but most of the content should be applicable to using C++/WinRT which is the supported way to write RNW apps.

```

Comment:
I'm confused about this bullet as it relates to the native modules docs, which I thought could be written in "pure" C++. 

Meta: I'm not sure what the relationship between these docs and the native modules docs all-up, especially this section.

Reply:
Native modules allow you to access the underlying platform, which for UWP you would do via cppwinrt. Yes, technically you can write a pure C++ module without calling any winrt components (e.g. the sample add x+y native module), but that's not what people will usually do.

The point of this paragraph is about "bring some library from nuget". We've seen people ask this on the repo, "can I use this win32 library in the context of a native module?", or they try it and it doesn't work. The info here is to warn them about it and provide some paths forward.